napari/utils/events/evented_model.py

```diff
@@ -181,37 +192,137 @@ class Config:
         # False, RuntimeError will be raised on model declaration
         arbitrary_types_allowed = True
         # whether to perform validation on assignment to attributes
+        # note that this should stay True for Field(allow_mutation) to work.
         validate_assignment = True
         # whether to treat any underscore non-class var attrs as private
         # https://pydantic-docs.helpmanual.io/usage/models/#private-model-attributes
         underscore_attrs_are_private = True
         # whether to validate field defaults (default: False)
-        # see https://github.com/napari/napari/pull/4138 before changing.
-        validate_all = False
+        validate_all = True
         # https://pydantic-docs.helpmanual.io/usage/exporting_models/#modeljson
         # NOTE: json_encoders are also added EventedMetaclass.__new__ if the
         # field declares a _json_encode method.
         json_encoders = _BASE_JSON_ENCODERS
+        # this custom use of allow mutation behaves as normal (0 is false, 1 and 2 are true)
+        # but allows us to use 2 as a special value meaning "inplace mutation"
+        allow_mutation = 2
```

Comment:
I don't love the special `2` value here. Is this really essential? Could we add a new property instead that could take a string?

Reply:
Not essential. Initially I had used an `extra` field in the `Field`, assuming `True` by default and allowing opting out. It made checks more clunky and verbose (I had to do stuff like `if model.__config__.allow_mutation and model.__config__.get('inplace_mutation', True) and field.allow_mutation and field.extra.get('inplace_mutation', True)`) but in practice it's doable.